### PR TITLE
Upgrade `meshery-operator`

### PR DESCRIPTION
**Pull request by acald-creator**
**Description**

This PR updates `meshery-operator` to use the latest `controller-runtime` at `v0.15.0`, which uses `k8s.io` `v0.27`.

This also updates the test to utilize `&client.SubResourceUpdateOptions{}` instead of `&client.UpdateOptions{}` as this is wrapped in `&client.SubResourceUpdateOptions{}`

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: api/v1alpha1/meshsync_types_test.go [hunk 1]
```diff
@@ -110,7 +110,9 @@ var _ = Describe("The test case for the meshsync CRDs", func() {
 				},
 			}
 			By("Update the status of the meshsync CRDs")
-			err := fakeClient.Status().Update(context, meshSync, &client.UpdateOptions{FieldManager: FileManager})
+			err := fakeClient.Status().Update(context, meshSync, &client.SubResourceUpdateOptions{
```
Comment (Aisuko) [inline api/v1alpha1/meshsync_types_test.go:113]: Hi, @acald-creator thanks for your contribution. Please check the CI issue first. I consider the test case "The meshsync CRDs update the status of the resources" is useful, I mean do we need to check the status?

**Review by Aisuko — CHANGES_REQUESTED** [review]

File: api/v1alpha1/meshsync_types_test.go (hunk 1 above)
Comment (acald-creator) [reply api/v1alpha1/meshsync_types_test.go:113]: @Aisuko I saw the CI issue, I will check to see if I can get this resolved.

**Review by leecalcote — APPROVED** [review]
Outside of the failing test... 👍

**Review by Azanul — APPROVED** [review]
LGTM (other than the failing test)

**Comment by nebula-aac** [comment]
bump, still working on this

**Comment by leecalcote** [comment]
Good, good.

**Comment by leecalcote** [comment]
@acald-creator @nebula-aac it'd be good to have this finished off

**Comment by nebula-aac** [comment]
@leecalcote working on it, but still trying to get the test case to past

locally, it runs fine

**Comment by leecalcote** [comment]
merge conflict

**Comment by Aisuko** [comment]
This already done by #536 , So, we close this one.

**Pull request closed without merging**